### PR TITLE
Document torch.quantile interpolation kwarg

### PR DESCRIPTION
**Pull request by heitorschueroff**
This PR documents the interpolation kwarg parameter added in https://github.com/pytorch/pytorch/pull/49267. Now that the forward compatibility period is over, we can expose this parameter.

### CODE REVIEW

File: aten/src/ATen/native/Sorting.cpp [hunk 1]
```diff
@@ -519,8 +518,7 @@ Tensor& quantile_out(
       out,
       self,
       q,
-      // NOLINTNEXTLINE(performance-move-const-arg)
-      std::move(dim),
+      dim,
```
Comment (mruberry) [inline aten/src/ATen/native/Sorting.cpp:521]: Why change this from a move?

File: test/test_reductions.py [hunk 2]
```diff
@@ -18,6 +17,9 @@
     instantiate_device_type_tests, onlyCPU, dtypes, dtypesIfCUDA, dtypesIfCPU,
     onlyOnCPUAndCUDA, onlyCUDA, largeTensorTest, precisionOverride)
 
+if TEST_SCIPY:
```
Comment (mruberry) [inline test/test_reductions.py:20]: Nice fix

File: torch/_torch_docs.py [hunk 3]
```diff
@@ -5771,10 +5778,22 @@ def merge_dicts(*dicts):
     >>> a = torch.arange(4.)
     >>> a
     tensor([0., 1., 2., 3.])
+    >>> torch.quantile(a, 0.6, interpolation='linear')
```
Comment (mruberry) [inline torch/_torch_docs.py:5781]: Nice examples

File: torch/_torch_docs.py [hunk 4]
```diff
@@ -5771,10 +5778,22 @@ def merge_dicts(*dicts):
...
+    tensor(2.)
+    >>> torch.quantile(a, 0.4, interpolation='nearest')
+    tensor(1.)
 """.format(**single_dim_common))
 
 add_docstr(torch.nanquantile, r"""
-nanquantile(input, q, dim=None, keepdim=False, *, out=None) -> Tensor
+nanquantile(input, q, dim=None, keepdim=False, *, interpolation='linear', out=None) -> Tensor
 
 This is a variant of :func:`torch.quantile` that "ignores" ``NaN`` values,
```
Comment (mruberry) [inline torch/_torch_docs.py:5798]: Something to think about when implementing additional nan* operators is if they should just specify how they treat nan's different and then point to the original function's documentation as if they're an alias

**Review by mruberry — APPROVED** [review]
Cool! This is a great follow-up.

Are the changes to moving dim necessary?

File: aten/src/ATen/native/Sorting.cpp (hunk 1 above)
Comment (heitorschueroff) [reply aten/src/ATen/native/Sorting.cpp:521]: dim is trivially copyable here so std::move has no effect and removing it also cleans up the NOLINT comments.

File: torch/_torch_docs.py (hunk 4 above)
Comment (heitorschueroff) [reply torch/_torch_docs.py:5798]: Good point!

**Review by cccclai — APPROVED** [review]
From pytorch edge side, prod models look good

**Comment by matthewfeickert** [comment]
Hi @heitorschueroff, @mruberry, @cccclai :wave:. As this was approved back in October, can you comment a bit on if you expect this PR to move forward (once conflicts are resolved) and (if possible) also comment on the feasibility of having this make its way into a `v1.10.X` release? It would be quite nice to have this `interpolation` kwarg in.

**Comment by anjali411** [comment]
I'll merge this in https://github.com/pytorch/pytorch/pull/70637

**Pull request closed without merging**